Simplifying setup.sh

nupplaphil opened the pull request:
FQDN is now used to create default base DN and default domain for postmaster.

Part of #42

nupplaphil committed:
Simplifying setup.sh

nupplaphil committed:
Restore LDAP_BASE_DN question (in case it differs from the FQDN)

nupplaphil committed:
fix intent

nupplaphil committed:
Adding timezone guess

nupplaphil committed:
Auto setup for bundled environment

nupplaphil commented:
- The guessed base DN is now based on the FQDN
- The timezone is guessed based on the local system
- If the default LDAP (= bundled) server is used, the PW is now random
- If the default DB (= bundeld) is used, the PW is now random

fbartels commented:
Thanks for your pr. I will need a bit of time to go through the changes, since I am on holiday until Saturday and do not have a full system to play with it. Will check back next weekend.

fbartels commented on setup.sh:
```diff
@@ -24,63 +32,86 @@ if [ ! -e ./.env ]; then
+		LDAP_BIND_PW=${new_value:-$value_default}
+	else
+		LDAP_ADMIN_PASSWORD=$(random_string)
+		LDAP_SEARCH_BASE="$LDAP_BIND_DN"
```
when just entering the setup, the ldap_search_base is empty for me.

fbartels replied:
it should be LDAP_BASE_DN instead of LDAP_BIND_DN

fbartels commented on setup.sh:
```diff
@@ -24,63 +32,86 @@ if [ ! -e ./.env ]; then
 	value_default="kopano.demo"
 	read -p "FQDN to be used (for reverse proxy) [$value_default]: " new_value
 	FQDN=${new_value:-$value_default}
+	LDAP_BASE_DN=$(fqdn_to_dn $FQDN)
```
move this down just above line 41?

fbartels commented on setup.sh:
```diff
@@ -24,63 +32,86 @@ if [ ! -e ./.env ]; then
-	LDAP_BIND_PW=${new_value:-$value_default}
+	if [ "$LDAP_SERVER" != "$value_default" ]; then
+		value_default="kopano123"
+		read -p "Password of the admin user [$value_default]: " new_value
```
When not using the bundled ldap then we do not have to differentiate between  admin/read-only user. remove the ldap_admin_password option.

fbartels commented on setup.sh:
```diff
@@ -24,63 +32,86 @@ if [ ! -e ./.env ]; then
-
+	if [ "$MYSQL_HOST" != "$value_default" ]; then
+		value_default="kopano123"
+		read -p "Password for the MySQL root user [$value_default]: " new_value
```
here its the same as with the ldap image. root user not needed, we just need a user to query the database (and potentially create tables)

fbartels commented:
Maybe at the end of setup.sh print something like "hey, you specified a different ldap/db than what is bundled with compose. Remember to remove these from the compose file"

fbartels reviewed (changes requested):
obligatory comment to mark this pr as "request changes"

nupplaphil committed:
Improving setup.sh
- Bugfixing $LDAP_BIND_DN
- Removing ADMIN-PWs for DB/LDAP in case of alternative server
- Warning message in case of not using the bundled DB/LDAP

nupplaphil commented:
I had to rebase because of missing changes in the setup

fbartels commented:
Thanks looks good now. I saw a smaller thing in the casing of the ldap dns, will fix that after merge myself.
